src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,99 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int elementPosition = 0;
+    private T[] elementData;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
 
+        if (elementPosition == elementData.length) {
+            resize();
+        }
+        elementData[elementPosition] = value;
+        elementPosition++;
+
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > elementPosition || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("ArrayList index out of bound");
+        }
+        if (elementPosition == elementData.length) {
+            resize();
+        }
+        System.arraycopy(elementData, index, elementData, index + 1, elementPosition - index);
+        elementData[index] = value;
+        elementPosition++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= elementPosition || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("ArrayList index out of bound");
+        }
```

Comment:
lets move this duplicated logic to separate method `checkIndex` or smth like this